contracts/factory/src/contract.rs

```diff
@@ -275,6 +277,78 @@ impl FactoryTrait for Factory {
             .get(&DataKey::Config)
             .expect("Factory: No multihop present in storage")
     }
+
+    fn get_user_portfolio(env: Env, sender: Address) -> UserPortfolio {
+        let initialized_pools = get_lp_vec(&env);
+        let mut lp_portfolio: Vec<LpPortfolio> = Vec::new(&env);
+        let mut stake_portfolio: Vec<StakePortfolio> = Vec::new(&env);
+
+        for address in initialized_pools {
+            let response: LiquidityPoolInfo = env.invoke_contract(
+                &address,
+                &Symbol::new(&env, "query_pool_info_for_factory"),
+                Vec::new(&env),
+            );
+
+            let asset_addr = response.pool_response.asset_a.address.clone();
+            let bsset_addr = response.pool_response.asset_b.address.clone();
+
+            let (asset_amount_available_for_user, bsset_amount_available_for_user) =
+                get_token_amount_per_liquidity_pool(&env, &asset_addr, &bsset_addr, &sender);
```

Comment:
This is wrong. You are providing current user's balance as amount of tokens in the liquidity pool?

Reply:
deleted